### ISSUE TITLE
sigal_set causes segmentation fault

### ISSUE THREAD

**Issue by RogerLaw** [shin1m/alexaagent#2]
If this line set the signal:
boost::asio::signal_set signals(*server.io_services().front(), SIGINT);
The program will crash upon any server requesting on http://localhost:3000

**Comment by shin1m** [comment]
Thank you for reporting.
I have pushed an SSL/TLS related fix.
I'm not sure if it has something to do with this issue.
Can you connect the server with https://localhost:3000?
Best regards,

**Comment by RogerLaw** [comment]
If signal_set is removed, I can access https://localhost:3000 and everything work fine. If this line presents, the program will crash and unable to load the webpage.

**Comment by shin1m** [comment]
I'm sorry, but I have not been able to reproduce it yet.
Could you give some details about it such as crash messages, library versions, and so on?

I have tested with:
- Arch Linux x86_64
- gcc 6.1.1
- boost 1.60.0
- nghttp2 1.13.0

Thank you.

**Comment by RogerLaw** [comment]
I am running on:
- Ubuntu Linux 4.4.0-31-generic x86_64
- GCC 5.3.1
- Boost 1.61.0
- nghttp2 1.13.0

Ctrl+C will terminate the app gracefully. however, when I open https://localhost:3000, its crashed with a: Segmentation fault (core dumped).

I tried using an individual ioservice instance for signal_set and no luck.

Segmentation fault (core dumped)

**Comment by shin1m** [comment]
Thank you for the information.
I will investigate it further, but give me some time to do so.

Although I'm not sure at all, disabling optimization might help.
Try './configure --enable-debug' or editing AM_CXXFLAGS in Makefile.am when you have time.

Best regards,

**Comment by RogerLaw** [comment]
I suppose the problem relates to my boost version. I will try a lower version later. Anyway it's not a big issue and I got work around. I copied the crash dumps here, just FYI. 

**\* Error in `./alexaagent': munmap_chunk(): invalid pointer: 0x000000000043d526 ***
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x77725)[0x7f5641b31725]
/lib/x86_64-linux-gnu/libc.so.6(cfree+0x1a8)[0x7f5641b3dc18]
/lib/x86_64-linux-gnu/libcrypto.so.1.0.0(CRYPTO_free+0x1d)[0x7f564307e2cd]
/lib/x86_64-linux-gnu/libcrypto.so.1.0.0(lh_free+0x39)[0x7f5643137b09]
/lib/x86_64-linux-gnu/libssl.so.1.0.0(SSL_CTX_free+0x7c)[0x7f56434b0cbc]
/lib/x86_64-linux-gnu/libssl.so.1.0.0(SSL_free+0x1fd)[0x7f56434b306d]
/lib/x86_64-linux-gnu/libssl.so.1.0.0(SSL_new+0x3a8)[0x7f56434b3468]
/usr/local/lib/libnghttp2_asio.so.1(_ZN7nghttp210asio_http26server6server12start_acceptERN5boost4asio3ssl7contextERNS4_21basic_socket_acceptorINS4_2ip3tcpENS4_23socket_acceptor_serviceISA_EEEERNS1_9serve_muxE+0xd6)[0x7f5642dc68f6]
/usr/local/lib/libnghttp2_asio.so.1(+0x61308)[0x7f5642dc9308]
./alexaagent[0x43c9de]
./alexaagent(_ZN5boost4asio6detail13epoll_reactor16descriptor_state11do_completeEPNS1_15task_io_serviceEPNS1_25task_io_service_operationERKNS_6system10error_codeEm+0x63)[0x43edb3]
./alexaagent[0x43c9de]
./alexaagent[0x43fb27]
./alexaagent(_ZN5boost4asio6detail15task_io_service3runERNS_6system10error_codeE+0xf1)[0x43f3ff]
./alexaagent(_ZN5boost4asio10io_service3runEv+0x3e)[0x43fe3e]
/usr/local/lib/libnghttp2_asio.so.1(_ZNSt17_Function_handlerIFSt10unique_ptrINSt13__future_base12_Result_baseENS2_8_DeleterEEvENS1_12_Task_setterIS0_INS1_7_ResultImEES3_ESt12_Bind_simpleIFSt7_Mem_fnIMN5boost4asio10io_serviceEFmvEESt10shared_ptrISE_EEEmEEE9_M_invokeERKSt9_Any_data+0x3d)[0x7f5642dbd38d]
/usr/local/lib/libnghttp2_asio.so.1(_ZNSt13__future_base13_State_baseV29_M_do_setEPSt8functionIFSt10unique_ptrINS_12_Result_baseENS3_8_DeleterEEvEEPb+0x29)[0x7f5642dbd4b9]
/lib/x86_64-linux-gnu/libpthread.so.0(+0xead9)[0x7f5642429ad9]
/usr/local/lib/libnghttp2_asio.so.1(_ZNSt6thread5_ImplISt12_Bind_simpleIFZNSt13__future_base17_Async_state_implIS1_IFSt7_Mem_fnIMN5boost4asio10io_serviceEFmvEESt10shared_ptrIS7_EEEmEC4EOSE_EUlvE_vEEE6_M_runEv+0xcd)[0x7f5642dbf72d]
/usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0xb8c70)[0x7f5642151c70]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x76fa)[0x7f56424226fa]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x6d)[0x7f5641bc0b5d]
======= Memory map: ========
00400000-0061f000 r-xp 00000000 08:10 1572987                            /media/playground/alexaagent/alexaagent
0081e000-0081f000 r--p 0021e000 08:10 1572987                            /media/playground/alexaagent/alexaagent
0081f000-00821000 rw-p 0021f000 08:10 1572987                            /media/playground/alexaagent/alexaagent
00821000-00822000 rw-p 00000000 00:00 0 
00843000-00909000 rw-p 00000000 00:00 0                                  [heap]
7f55f4000000-7f55f4021000 rw-p 00000000 00:00 0 
7f55f4021000-7f55f8000000 ---p 00000000 00:00 0 
7f55fc000000-7f55fc021000 rw-p 00000000 00:00 0 
7f55fc021000-7f5600000000 ---p 00000000 00:00 0 
7f5608000000-7f5608021000 rw-p 00000000 00:00 0 
7f5608021000-7f560c000000 ---p 00000000 00:00 0 
7f5614000000-7f5614021000 rw-p 00000000 00:00 0 
7f5614021000-7f5618000000 ---p 00000000 00:00 0 
7f5618000000-7f561843f000 rw-p 00000000 00:00 0 
7f561843f000-7f561c000000 ---p 00000000 00:00 0 
7f561c000000-7f561c021000 rw-p 00000000 00:00 0 
7f561c021000-7f5620000000 ---p 00000000 00:00 0 
7f5622ffd000-7f5622ffe000 ---p 00000000 00:00 0 
7f5622ffe000-7f56237fe000 rw-p 00000000 00:00 0 
7f56237fe000-7f56237ff000 ---p 00000000 00:00 0 
7f56237ff000-7f5623fff000 rw-p 00000000 00:00 0 
7f5623fff000-7f5628000000 rw-s 00000000 00:13 5                          /dev/shm/pulse-shm-453702293
7f5628000000-7f5628021000 rw-p 00000000 00:00 0 
7f5628021000-7f562c000000 ---p 00000000 00:00 0 
7f562c5bf000-7f562c5c0000 ---p 00000000 00:00 0 
7f562c5c0000-7f562cdc0000 rw-p 00000000 00:00 0 
7f562cdc0000-7f562cdc5000 r-xp 00000000 08:01 530007                     /lib/x86_64-linux-gnu/libnss_dns-2.23.so
7f562cdc5000-7f562cfc5000 ---p 00005000 08:01 530007                     /lib/x86_64-linux-gnu/libnss_dns-2.23.so
7f562cfc5000-7f562cfc6000 r--p 00005000 08:01 530007                     /lib/x86_64-linux-gnu/libnss_dns-2.23.so
7f562cfc6000-7f562cfc7000 rw-p 00006000 08:01 530007                     /lib/x86_64-linux-gnu/libnss_dns-2.23.so
7f562cfc7000-7f562cfc9000 r-xp 00000000 08:01 530015                     /lib/x86_64-linux-gnu/libnss_mdns4_minimal.so.2
7f562cfc9000-7f562d1c8000 ---p 00002000 08:01 530015                     /lib/x86_64-linux-gnu/libnss_mdns4_minimal.so.2
7f562d1c8000-7f562d1c9000 r--p 00001000 08:01 530015                     /lib/x86_64-linux-gnu/libnss_mdns4_minimal.so.2
7f562d1c9000-7f562d1ca000 rw-p 00002000 08:01 530015                     /lib/x86_64-linux-gnu/libnss_mdns4_minimal.so.2
7f562d1ca000-7f562d1cb000 ---p 00000000 00:00 0 
7f562d1cb000-7f562d9cb000 rw-p 00000000 00:00 0 
7f562d9cb000-7f562d9cc000 ---p 00000000 00:00 0 
7f562d9cc000-7f562e1cc000 rw-p 00000000 00:00 0 
7f562e1cc000-7f562e1d7000 r-xp 00000000 08:01 530009                     /lib/x86_64-linux-gnu/libnss_files-2.23.so
7f562e1d7000-7f562e3d6000 ---p 0000b000 08:01 530009                     /lib/x86_64-linux-gnu/libnss_files-2.23.so
7f562e3d6000-7f562e3d7000 r--p 0000a000 08:01 530009                     /lib/x86_64-linux-gnu/libnss_files-2.23.so
7f562e3d7000-7f562e3d8000 rw-p 0000b000 08:01 530009                     /lib/x86_64-linux-gnu/libnss_files-2.23.so
7f562e3d8000-7f562ebf6000 rw-p 00000000 00:00 0 
7f562ebf6000-7f562ebf7000 ---p 00000000 00:00 0 
7f562ebf7000-7f562f3f7000 rw-p 00000000 00:00 0 
7f562f3f7000-7f56333f8000 rw-s 00000000 00:13 17                         /dev/shm/pulse-shm-2672117603
7f56333f8000-7f56333fe000 r-xp 00000000 08:01 3569                       /usr/lib/x86_64-linux-gnu/alsa-lib/libasound_module_pcm_pulse.so
7f56333fe000-7f56335fd000 ---p 00006000 08:01 3569                       /usr/lib/x86_64-linux-gnu/alsa-lib/libasound_module_pcm_pulse.so
7f56335fd000-7f56335fe000 r--p 00005000 08:01 3569                       /usr/lib/x86_64-linux-gnu/alsa-lib/libasound_module_pcm_pulse.so
7f56335fe000-7f56335ff000 rw-p 00006000 08:01 3569                       /usr/lib/x86_64-linux-gnu/alsa-lib/libasound_module_pcm_pulse.so
7f56335ff000-7f563366d000 r-xp 00000000 08:01 530039                     /lib/x86_64-linux-gnu/libpcre.so.3.13.2
7f563366d000-7f563386d000 ---p 0006e000 08:01 530039                     /lib/x86_64-linux-gnu/libpcre.so.3.13.2
7f563386d000-7f563386e000 r--p 0006e000 08:01 530039                     /lib/x86_64-linux-gnu/libpcre.so.3.13.2
7f563386e000-7f563386f000 rw-p 0006f000 08:01 530039                     /lib/x86_64-linux-gnu/libpcre.so.3.13.2
7f563386f000-7f56338e2000 r-xp 00000000 08:01 795521                     /usr/lib/x86_64-linux-gnu/libFLAC.so.8.3.0
7f56338e2000-7f5633ae2000 ---p 00073000 08:01 795521                     /usr/lib/x86_64-linux-gnu/libFLAC.so.8.3.0
7f5633ae2000-7f5633ae3000 r--p 00073000 08:01 795521                     /usr/lib/x86_64-linux-gnu/libFLAC.so.8.3.0
7f5633ae3000-7f5633ae4000 rw-p 00074000 08:01 795521                     /usr/lib/x86_64-linux-gnu/libFLAC.so.8.3.0
7f5633ae4000-7f5633afa000 r-xp 00000000 08:01 530003                     /lib/x86_64-linux-gnu/libnsl-2.23.so
7f5633afa000-7f5633cf9000 ---p 00016000 08:01 530003                     /lib/x86_64-linux-gnu/libnsl-2.23.so
7f5633cf9000-7f5633cfa000 r--p 00015000 08:01 530003                     /lib/x86_64-linux-gnu/libnsl-2.23.so
7f5633cfa000-7f5633cfb000 rw-p 00016000 08:01 530003                     /lib/x86_64-linux-gnu/libnsl-2.23.so
7f5633cfb000-7f5633cfd000 rw-p 00000000 00:00 0 
7f5633cfd000-7f5633d1c000 r-xp 00000000 08:01 530068                     /lib/x86_64-linux-gnu/libselinux.so.1
7f5633d1c000-7f5633f1b000 ---p 0001f000 08:01 530068                     /lib/x86_64-linux-gnu/libselinux.so.1
7f5633f1b000-7f5633f1c000 r--p 0001e000 08:01 530068                     /lib/x86_64-linux-gnu/libselinux.so.1
7f5633f1c000-7f5633f1d000 rw-p 0001f000 08:01 530068                     /lib/x86_64-linux-gnu/libselinux.so.1
7f5633f1d000-7f5633f1f000 rw-p 00000000 00:00 0 
7f5633f1f000-7f5633f24000 r-xp 00000000 08:01 795680                     /usr/lib/x86_64-linux-gnu/libXdmcp.so.6.0.0
7f5633f24000-7f5634123000 ---p 00005000 08:01 795680                     /usr/lib/x86_64-linux-gnu/libXdmcp.so.6.0.0
7f5634123000-7f5634124000 r--p 00004000 08:01 795680                     /usr/lib/x86_64-linux-gnu/libXdmcp.so.6.0.0
7f5634124000-7f5634125000 rw-p 00005000 08:01 795680                     /usr/lib/x86_64-linux-gnu/libXdmcp.so.6.0.0
7f5634125000-7f5634127000 r-xp 00000000 08:01 795669                     /usr/lib/x86_64-linux-gnu/libXau.so.6.0.0
7f5634127000-7f5634327000 ---p 00002000 08:01 795669                     /usr/lib/x86_64-linux-gnu/libXau.so.6.0.0
7f5634327000-7f5634328000 r--p 00002000 08:01 795669                     /usr/lib/x86_64-linux-gnu/libXau.so.6.0.0
7f5634328000-7f5634329000 rw-p 00003000 08:01 795669                     /usr/lib/x86_64-linux-gnu/libXau.so.6.0.0
7f5634329000-7f563432e000 r-xp 00000000 08:01 795771                     /usr/lib/x86_64-linux-gnu/libasyncns.so.0.3.1
7f563432e000-7f563452d000 ---p 00005000 08:01 795771                     /usr/lib/x86_64-linux-gnu/libasyncns.so.0.3.1
7f563452d000-7f563452e000 r--p 00004000 08:01 795771                     /usr/lib/x86_64-linux-gnu/libasyncns.so.0.3.1
7f563452e000-7f563452f000 rw-p 00005000 08:01 795771                     /usr/lib/x86_64-linux-gnu/libasyncns.so.0.3.1
7f563452f000-7f5634591000 r-xp 00000000 08:01 796778                     /usr/lib/x86_64-linux-gnu/libsndfile.so.1.0.25
7f5634591000-7f5634791000 ---p 00062000 08:01 796778                     /usr/lib/x86_64-linux-gnu/libsndfile.so.1.0.25
7f5634791000-7f5634793000 r--p 00062000 08:01 796778                     /usr/lib/x86_64-linux-gnu/libsndfile.so.1.0.25
7f5634793000-7f5634794000 rw-p 00064000 08:01 796778                     /usr/lib/x86_64-linux-gnu/libsndfile.so.1.0.25
7f5634794000-7f5634798000 rw-p 00000000 00:00 0 
7f5634798000-7f56347a0000 r-xp 00000000 08:01 530096                     /lib/x86_64-linux-gnu/libwrap.so.0.7.6
7f56347a0000-7f563499f000 ---p 00008000 08:01 530096                     /lib/x86_64-linux-gnu/libwrap.so.0.7.6
7f563499f000-7f56349a0000 r--p 00007000 08:01 530096                     /lib/x86_64-linux-gnu/libwrap.so.0.7.6
7f56349a0000-7f56349a1000 rw-p 00008000 08:01 530096                     /lib/x86_64-linux-gnu/libwrap.so.0.7.6
7f56349a1000-7f56349a2000 rw-p 00000000 00:00 0 
7f56349a2000-7f56349c3000 r-xp 00000000 08:01 797047                     /usr/lib/x86_64-linux-gnu/libxcb.so.1.1.0
7f56349c3000-7f5634bc2000 ---p 00021000 08:01 797047                     /usr/lib/x86_64-linux-gnu/libxcb.so.1.1.0
7f5634bc2000-7f5634bc3000 r--p 00020000 08:01 797047                     /usr/lib/x86_64-linux-gnu/libxcb.so.1.1.0
7f5634bc3000-7f5634bc4000 rw-p 00021000 08:01 797047                     /usr/lib/x86_64-linux-gnu/libxcb.so.1.1.0
7f5634bc4000-7f5634c0e000 r-xp 00000000 08:01 529926                     /lib/x86_64-linux-gnu/libdbus-1.so.3.14.6
7f5634c0e000-7f5634e0e000 ---p 0004a000 08:01 529926                     /lib/x86_64-linux-gnu/libdbus-1.so.3.14.6
7f5634e0e000-7f5634e0f000 r--p 0004a000 08:01 529926                     /lib/x86_64-linux-gnu/libdbus-1.so.3.14.6
7f5634e0f000-7f5634e10000 rw-p 0004b000 08:01 529926                     /lib/x86_64-linux-gnu/libdbus-1.so.3.14.6
7f5634e10000-7f5634e89000 r-xp 00000000 08:01 5616                       /usr/lib/x86_64-linux-gnu/pulseaudio/libpulsecommon-8.0.so
7f5634e89000-7f5635088000 ---p 00079000 08:01 5616                       /usr/lib/x86_64-linux-gnu/pulseaudio/libpulsecommon-8.0.so
7f5635088000-7f5635089000 r--p 00078000 08:01 5616                       /usr/lib/x86_64-linux-gnu/pulseaudio/libpulsecommon-8.0.so
7f5635089000-7f563508a000 rw-p 00079000 08:01 5616                       /usr/lib/x86_64-linux-gnu/pulseaudio/libpulsecommon-8.0.so
7f563508a000-7f5635094000 r-xp 00000000 08:01 529969                     /lib/x86_64-linux-gnu/libjson-c.so.2.0.0
7f5635094000-7f5635293000 ---p 0000a000 08:01 529969                     /lib/x86_64-linux-gnu/libjson-c.so.2.0.0
7f5635293000-7f5635294000 r--p 00009000 08:01 529969                     /lib/x86_64-linux-gnu/libjson-c.so.2.0.0
7f5635294000-7f5635295000 rw-p 0000a000 08:01 529969                     /lib/x86_64-linux-gnu/libjson-c.so.2.0.0
7f5635295000-7f56352e3000 r-xp 00000000 08:01 796666                     /usr/lib/x86_64-linux-gnu/libpulse.so.0.19.0
7f56352e3000-7f56354e2000 ---p 0004e000 08:01 796666                     /usr/lib/x86_64-linux-gnu/libpulse.so.0.19.0
7f56354e2000-7f56354e3000 r--p 0004d000 08:01 796666                     /usr/lib/x86_64-linux-gnu/libpulse.so.0.19.0
7f56354e3000-7f56354e4000 rw-p 0004e000 08:01 796666                     /usr/lib/x86_64-linux-gnu/libpulse.so.0.19.0
7f5635530000-7f56356e6000 rw-p 00000000 00:00 0 
7f56356e6000-7f56357df000 r-xp 00000000 08:01 795761                     /usr/lib/x86_64-linux-gnu/libasound.so.2.0.0
7f56357df000-7f56359de000 ---p 000f9000 08:01 795761                     /usr/lib/x86_64-linux-gnu/libasound.so.2.0.0
7f56359de000-7f56359e5000 r--p 000f8000 08:01 795761                     /usr/lib/x86_64-linux-gnu/libasound.so.2.0.0
7f56359e5000-7f56359e6000 rw-p 000ff000 08:01 795761                     /usr/lib/x86_64-linux-gnu/libasound.so.2.0.0
7f56359e6000-7f56359fd000 r-xp 00000000 08:01 530062                     /lib/x86_64-linux-gnu/libresolv-2.23.so
7f56359fd000-7f5635bfd000 ---p 00017000 08:01 530062                     /lib/x86_64-linux-gnu/libresolv-2.23.so
7f5635bfd000-7f5635bfe000 r--p 00017000 08:01 530062                     /lib/x86_64-linux-gnu/libresolv-2.23.so
7f5635bfe000-7f5635bff000 rw-p 00018000 08:01 530062                     /lib/x86_64-linux-gnu/libresolv-2.23.so
7f5635bff000-7f5635c01000 rw-p 00000000 00:00 0 
7f5635c01000-7f5635c04000 r-xp 00000000 08:01 529971                     /lib/x86_64-linux-gnu/libkeyutils.so.1.5
7f5635c04000-7f5635e03000 ---p 00003000 08:01 529971                     /lib/x86_64-linux-gnu/libkeyutils.so.1.5
7f5635e03000-7f5635e04000 r--p 00002000 08:01 529971                     /lib/x86_64-linux-gnu/libkeyutils.so.1.5
7f5635e04000-7f5635e05000 rw-p 00003000 08:01 529971                     /lib/x86_64-linux-gnu/libkeyutils.so.1.5
7f5635e05000-7f5635e0c000 r-xp 00000000 08:01 796051                     /usr/lib/x86_64-linux-gnu/libffi.so.6.0.4
7f5635e0c000-7f563600b000 ---p 00007000 08:01 796051                     /usr/lib/x86_64-linux-gnu/libffi.so.6.0.4
7f563600b000-7f563600c000 r--p 00006000 08:01 796051                     /usr/lib/x86_64-linux-gnu/libffi.so.6.0.4
7f563600c000-7f563600d000 rw-p 00007000 08:01 796051                     /usr/lib/x86_64-linux-gnu/libffi.so.6.0.4
7f563600d000-7f5636033000 r-xp 00000000 08:01 530211                     /lib/x86_64-linux-gnu/libexpat.so.1.6.0
7f5636033000-7f5636233000 ---p 00026000 08:01 530211                     /lib/x86_64-linux-gnu/libexpat.so.1.6.0
7f5636233000-7f5636235000 r--p 00026000 08:01 530211                     /lib/x86_64-linux-gnu/libexpat.so.1.6.0
7f5636235000-7f5636236000 rw-p 00028000 08:01 530211                     /lib/x86_64-linux-gnu/libexpat.so.1.6.0
7f5636236000-7f5636240000 r-xp 00000000 08:01 796414                     /usr/lib/x86_64-linux-gnu/libkrb5support.so.0.1
7f5636240000-7f563643f000 ---p 0000a000 08:01 796414                     /usr/lib/x86_64-linux-gnu/libkrb5support.so.0.1
7f563643f000-7f5636440000 r--p 00009000 08:01 796414                     /usr/lib/x86_64-linux-gnu/libkrb5support.so.0.1
7f5636440000-7f5636441000 rw-p 0000a000 08:01 796414                     /usr/lib/x86_64-linux-gnu/libkrb5support.so.0.1
7f5636441000-7f5636444000 r-xp 00000000 08:01 529919                     /lib/x86_64-linux-gnu/libcom_err.so.2.1
7f5636444000-7f5636643000 ---p 00003000 08:01 529919                     /lib/x86_64-linux-gnu/libcom_err.so.2.1
7f5636643000-7f5636644000 r--p 00002000 08:01 529919                     /lib/x86_64-linux-gnu/libcom_err.so.2.1
7f5636644000-7f5636645000 rw-p 00003000 08:01 529919                     /lib/x86_64-linux-gnu/libcom_err.so.2.1
7f5636645000-7f5636671000 r-xp 00000000 08:01 796406                     /usr/lib/x86_64-linux-gnu/libk5crypto.so.3.1
7f5636671000-7f5636870000 ---p 0002c000 08:01 796406                     /usr/lib/x86_64-linux-gnu/libk5crypto.so.3.1
7f5636870000-7f5636872000 r--p 0002b000 08:01 796406                     /usr/lib/x86_64-linux-gnu/libk5crypto.so.3.1
7f5636872000-7f5636873000 rw-p 0002d000 08:01 796406                     /usr/lib/x86_64-linux-gnu/libk5crypto.so.3.1
7f5636873000-7f5636874000 rw-p 00000000 00:00 0 
7f5636874000-7f5636937000 r-xp 00000000 08:01 796412                     /usr/lib/x86_64-linux-gnu/libkrb5.so.3.3
7f5636937000-7f5636b37000 ---p 000c3000 08:01 796412                     /usr/lib/x86_64-linux-gnu/libkrb5.so.3.3
7f5636b37000-7f5636b44000 r--p 000c3000 08:01 796412                     /usr/lib/x86_64-linux-gnu/libkrb5.so.3.3
7f5636b44000-7f5636b46000 rw-p 000d0000 08:01 796412                     /usr/lib/x86_64-linux-gnu/libkrb5.so.3.3
7f5636b46000-7f5636b58000 r-xp 00000000 08:01 529954                     /lib/x86_64-linux-gnu/libgpg-error.so.0.17.0
7f5636b58000-7f5636d58000 ---p 00012000 08:01 529954                     /lib/x86_64-linux-gnu/libgpg-error.so.0.17.0
7f5636d58000-7f5636d59000 r--p 00012000 08:01 529954                     /lib/x86_64-linux-gnu/libgpg-error.so.0.17.0
7f5636d59000-7f5636d5a000 rw-p 00013000 08:01 529954                     /lib/x86_64-linux-gnu/libgpg-error.so.0.17.0
7f5636d5a000-7f5636d7b000 r-xp 00000000 08:01 790042                     /usr/lib/x86_64-linux-gnu/libgomp.so.1.0.0
7f5636d7b000-7f5636f7a000 ---p 00021000 08:01 790042                     /usr/lib/x86_64-linux-gnu/libgomp.so.1.0.0
7f5636f7a000-7f5636f7b000 r--p 00020000 08:01 790042                     /usr/lib/x86_64-linux-gnu/libgomp.so.1.0.0
7f5636f7b000-7f5636f7c000 rw-p 00021000 08:01 790042                     /usr/lib/x86_64-linux-gnu/libgomp.so.1.0.0
7f5636f7c000-7f5638832000 r-xp 00000000 08:01 796346                     /usr/lib/x86_64-linux-gnu/libicudata.so.55.1Aborted (core dumped)

**Comment by shin1m** [comment]
Thank you for the details.
I'm still investigating and have not been able to reproduce it yet.

I have installed Ubuntu 16.04.1 x86_64 on a virtual machine, and been able to run it successfully.
I prepared almost same versions as yours, but gcc is slightly different as it was automatically installed by default:
- Linux version 4.4.0-31-generic
- gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.1)
- Boost 1.61.0
- nghttp2 1.13.0

Best regards,
